source/extensions/clusters/aggregate/cluster.cc

```diff
@@ -46,13 +46,14 @@ AggregateClusterLoadBalancer::AggregateClusterLoadBalancer(
 
 void AggregateClusterLoadBalancer::addMemberUpdateCallbackForCluster(
     Upstream::ThreadLocalCluster& thread_local_cluster) {
-  thread_local_cluster.prioritySet().addMemberUpdateCb(
-      [this, target_cluster_info = thread_local_cluster.info()](const Upstream::HostVector&,
-                                                                const Upstream::HostVector&) {
-        ENVOY_LOG(debug, "member update for cluster '{}' in aggregate cluster '{}'",
-                  target_cluster_info->name(), parent_info_->name());
-        refresh();
-      });
+  member_update_cbs_[thread_local_cluster.info()->name()] =
```

Comment:
For this and other places that we now capture, were we leaking before, since there was never a remove()?

Reply:
Correct, if someone had delivered the aggregate cluster via CDS and then removed it, there would be crashes. I think this change has fixed multiple different latent bugs.